pkg/kubelet/config/flags.go

```diff
@@ -52,10 +52,8 @@ type ContainerRuntimeOptions struct {
 func (s *ContainerRuntimeOptions) AddFlags(fs *pflag.FlagSet) {
 	// General settings.
 	fs.StringVar(&s.ContainerRuntime, "container-runtime", s.ContainerRuntime, "The container runtime to use. Possible value: 'remote'.")
-	fs.MarkDeprecated("container-runtime", "will be removed in 1.27 as the only valid value is 'remote'")
 	fs.StringVar(&s.RuntimeCgroups, "runtime-cgroups", s.RuntimeCgroups, "Optional absolute name of cgroups to create and run the runtime in.")
 	fs.StringVar(&s.PodSandboxImage, "pod-infra-container-image", s.PodSandboxImage, fmt.Sprintf("Specified image will not be pruned by the image garbage collector. CRI implementations have their own configuration to set this image."))
-	fs.MarkDeprecated("pod-infra-container-image", "will be removed in 1.27. Image garbage collector will get sandbox image information from CRI.")
```

Comment:
Unfortunately this is not implemented yet on containerd: https://github.com/containerd/containerd/issues/6352

Perhaps instead of removing it, extend it for couple more releases

/hold

Reply:
Line 55 removing seems to be valid.

@SergeyKanzhelev should @calvin0327 keep `pod-infra-container-image` and remove the `container-runtime` in this PR?

Reply:
Yes

Reply:
thanks, I have recovered the `pod-infra-container-image`

Reply:
/unhold